contracts/LinearReverseDutchAuction.sol

```diff
@@ -0,0 +1,170 @@
+
+        auctions[auctionCounter] = AuctionData({
+            id: auctionCounter,
+            name:name,
+            description:description,
+            imgUrl:imgUrl,
+            auctioneer: msg.sender,
+            auctionType: auctionType,
+            auctionedToken: auctionedToken,
+            auctionedTokenIdOrAmount: auctionedTokenIdOrAmount,
+            biddingToken: biddingToken,
+            startingPrice: startingPrice,
+            availableFunds: 0,
+            reservedPrice: reservedPrice,
+            winner: msg.sender,
+            deadline: deadline,
+            duration: duration,
+            isClaimed: false
+        });
+
+        emit AuctionCreated(
+            auctionCounter++,
+            name,
+            description,
+            imgUrl,
+            msg.sender,
+            auctionType,
+            auctionedToken,
+            auctionedTokenIdOrAmount,
+            biddingToken,
+            startingPrice,
+            reservedPrice,
+            deadline
+        );
+    }
+
+    function getCurrentPrice(uint256 auctionId) public view validAuctionId(auctionId) returns (uint256) {
+        AuctionData storage auction=auctions[auctionId];
+        require(block.timestamp<auction.deadline,"Auction has ended");
+        require(!auction.isClaimed,"Auction has ended");
+
+        // price(t) = startingPrice - (((startingPrice - reservedPrice) * (timeElapsed)) / duration)
+        return auction.startingPrice - (((auction.startingPrice-auction.reservedPrice)*(block.timestamp - (auction.deadline - auction.duration)))/auction.duration);
+    }
+
+    //placeBid function is not required in this auction type as the price is determined by the auctioneer and the auctioneer can withdraw the item on placing the bid only
+    function withdrawItem(uint256 auctionId,uint256 bidAmount) external validAuctionId(auctionId) {
+        AuctionData storage auction = auctions[auctionId];
+        require(block.timestamp<auction.deadline,"Auction has ended");
+        require(!auction.isClaimed,"Auction has been settled");
+
+        uint256 currentPrice=getCurrentPrice(auctionId);
+        require(bidAmount>=currentPrice,"Bid amount is less than current price");
+        
+        SafeERC20.safeTransferFrom(IERC20(auction.biddingToken),msg.sender,address(this),currentPrice);
+        if(auction.auctionType == AuctionType.NFT){
+            IERC721(auction.auctionedToken).safeTransferFrom(address(this),msg.sender,auction.auctionedTokenIdOrAmount);
+        }else{
+            SafeERC20.safeTransfer(IERC20(auction.auctionedToken),msg.sender,auction.auctionedTokenIdOrAmount);
+        }
+
+        auction.winner=msg.sender;
+        auction.availableFunds=bidAmount;
+        auction.isClaimed=true;
+
+        emit itemWithdrawn(auctionId,msg.sender,auction.auctionedToken,auction.auctionedTokenIdOrAmount);
+    }
+
+    function withdrawFunds(uint256 auctionId) external validAuctionId(auctionId){
```

Comment:
I think the functions `withdrawFunds` from most auctions can be moved to the base `Auction` contract.